python/ql/lib/semmle/python/dataflow/new/internal/DataFlowDispatch.qll

```diff
@@ -460,10 +460,26 @@ private CallCfgNode getSuperCall() {
   )
 }
 
+/**
+ * Holds if the file `f` should be ignored when computing the call-graph.
+ *
+ * We currently see a performance problem when analyzing the `sympy` PyPI package,
+ * which can be part of the database when dependencies are installed and extracted.
+ * From what we can understand, SymPy is using Python in a exotic way, so the fact that
+ * our analysis currently does not handle this project has nothing to say about our
+ * ability to handle normal Python code. Furthermore, SymPy does not look to be relevant
+ * in a security context, so we should not loose out on any security results by doing
+ * this.
+ */
+private predicate ignoreForCallGraph(File f) {
```

Comment:
Maybe it would make more sense for this to be a predicate on `Node`s? That would save the repeated `.getLocation().getFile()` and make the interface a bit more flexible. After all, we might want to ignore things on a more fine-grained basis in the future.

Reply:
I didn't put too much thought into making the perfect interface for this. If we need it to be more flexible to, I don't see anything that would prevent us from just altering it in the future. I'd like to keep it at ignoring files for now, since it's more upfront about what things are being ignored, and hopefully (:crossed_fingers:) we will never need to ignore specific nodes.

Reply:
Fair enough. 🙂